### PR TITLE
Homework 17

### PR DESCRIPTION
**Pull request by vladimirchizhenkov**

### CODE REVIEW

File: homework_17/jest-test/calculator.spec.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import { Calculator } from '../src/calculator';
+import { operation } from "../src/types";
+
+describe('Jest tests for calculator:', () => {
+    //Не получится установить тип Calculator, т.к. в конце тестов переменная calculator принимает значение null.
```
Comment (VladSelikh) [inline homework_17/jest-test/calculator.spec.ts:5]: Please delete this comment.

File: homework_17/src/calculator.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { operation } from './types';
+
+export class Calculator {
+    typeOperation: operation;
+
+    constructor(typeOperation: operation) {
+        this.typeOperation = typeOperation;
```
Comment (VladSelikh) [inline homework_17/src/calculator.ts:7]: Why do you need to pass typeOperation to the `constructor`? I suppose your `calculator` will work correctly with empty constructor. Please delete this variable.

File: homework_17/jest-test/calculator.spec.ts (hunk 1 above)
Comment (vladimirchizhenkov) [reply homework_17/jest-test/calculator.spec.ts]: fix.

File: homework_17/src/calculator.ts (hunk 2 above)
Comment (vladimirchizhenkov) [reply homework_17/src/calculator.ts]: fixed.

File: homework_17/mocha-test/calculator.spec.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import { Calculator } from '../src/calculator';
+import { ErrorReporter } from '../src/error-reporter';
+import { expect } from 'chai';
+
+import { operation } from '../src/types';
+
```
Comment (VladSelikh) [inline homework_17/mocha-test/calculator.spec.ts:6]: Please use object destructure:
```
const { sum, division, multiplication, subtraction } = operation;
..//..
it('Should sum two positive numbers', () => {
            expect(calculator.sum(10, 12)).to.eq(22, ErrorReporter.report(sum));
        });
```

File: homework_17/src/error-reporter.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export class ErrorReporter {
+    public static report = (typeOperation: string) => `Wrong ${typeOperation} of numbers`;
```
Comment (VladSelikh) [inline homework_17/src/error-reporter.ts:2]: Please be more specific in the type of `typeOperation`:
`public static report = (typeOperation: operation) => `Wrong ${typeOperation} of numbers`;`

File: homework_17/mocha-test/calculator.spec.ts (hunk 3 above)
Comment (vladimirchizhenkov) [reply homework_17/mocha-test/calculator.spec.ts:6]: fixed.

File: homework_17/src/error-reporter.ts (hunk 4 above)
Comment (vladimirchizhenkov) [reply homework_17/src/error-reporter.ts]: fixed

**Review by VladSelikh — APPROVED** [review]

**Pull request merged**